config/tests.config.js

```diff
@@ -1,27 +1,45 @@
 const path = require('path');
 const glob = require('glob');
+const WebpackShellPlugin = require('webpack-shell-plugin');
 
 module.exports = cfg => {
+  const onFail = process.env.ENABLE_WATCH === 'true' ? '|| true' : '';
+  let args = '--no-coverage';
+
   // Include all .test.js files as entry points
   glob.sync(path.resolve(__dirname, '../src/**/*.test.js')).map(f => path.relative(__dirname, f).replace('..', '.')).forEach(f => {
     cfg.entry[path.basename(f, '.js')] = [f];
   });
 
   // Add instrumentation for code coverage checking
-  cfg.module.rules.unshift({
-    test: /\.(js)$/,
-    use: [
-      {
-        loader: 'istanbul-instrumenter-loader',
-        options: {
-          esModules: true,
-          produceSourceMap: true
+  if (process.env.ENABLE_COVERAGE === 'true') {
+    cfg.module.rules.unshift({
+      test: /\.(js)$/,
+      use: [
+        {
+          loader: 'istanbul-instrumenter-loader',
+          options: {
+            esModules: true,
+            produceSourceMap: true
+          }
         }
-      }
-    ],
-    enforce: 'post',
-    exclude: /node_modules|\.tests\.js$/
-  });
+      ],
+      enforce: 'post',
+      exclude: /node_modules|\.tests\.js$/
+    });
+    args = `--100`;
+  }
+
+  // Add plugin to execute test runner
+  cfg.plugins.push(new WebpackShellPlugin({
+    onBuildExit: [{
+      command: `sh`,
+      args: [
+        `-c`,
+        `./node_modules/.bin/tap ${args} --nyc-arg='--instrument=false' --reporter classic -J ./dist/*.test.js && echo '\n✓ ALL TESTS PASSED!' ${onFail}`
```

Comment:
Why are we forcing a success?

Reply:
when running in watch mode ... you want the tests to run after each time the source changes ... if this script returns with a failure status then it will cause the webpack watch process to exit, which means you'd have to restart the process every time you cause a test failure which kinda defeats the point.

When running _without_ the watch loop we _do_ want the tests to exit with a nasty status to ensure we get test failures.

Reply:
Sounds good! :tada: